src/Accounts/Accounts/Models/SubscriptionClientProxy.cs

```diff
@@ -0,0 +1,190 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Rest.Azure;
+using System;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System.Threading;
+using SubscriptionClientVersion2016 = Microsoft.Azure.Commands.ResourceManager.Common.Utilities.SubscriptionClientWrapper;
+using SubscriptionClientVersion2019 = Microsoft.Azure.Commands.ResourceManager.Common.Utilities.Version2019_06_01.SubscriptionClientWrapper;
+
+namespace Microsoft.Azure.Commands.Profile.Models
+{
+    //Use queue to handle the subscription clients priority
+    class SubscritpionClientCandidates : ConcurrentQueue<ISubscriptionClientWrapper>
+    {
+        static ReaderWriterLockSlim instanceLock = new ReaderWriterLockSlim(LockRecursionPolicy.SupportsRecursion);
+        static private SubscritpionClientCandidates _instance = null;
+        static public SubscritpionClientCandidates Instance
```

Comment:
nit: please keep the code style consistent, e.g. always add access modifier for all members